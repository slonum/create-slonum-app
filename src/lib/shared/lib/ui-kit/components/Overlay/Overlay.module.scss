.overlay {
  display: block;
  position: fixed;
  top: 0;
  left: 0; 
  width: 100vw;
  height: 100vh;
  z-index: 10008;
  background: rgba(31, 29, 42, 0.7);
  cursor: pointer;
  animation: popup-appearance .2s;
}

@keyframes popup-appearance {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

.story {
  display: flex;
  align-items: center;
  justify-content: center;
  position: static;
}