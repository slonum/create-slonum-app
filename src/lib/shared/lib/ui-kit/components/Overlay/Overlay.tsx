import cl from 'classnames';
import { FC, useEffect } from 'react';
import styles from './Overlay.module.scss';

interface IOverlayProps {
  isActive: boolean;
  close: () => void;
  className?: string;
}

export const Overlay: FC<IOverlayProps> = ({ isActive, close, className }) => {
  useEffect(() => {
    if (!isActive) {
      document.body.style.overflow = '';
    } else {
      document.body.style.overflow = 'hidden';
    }
  }, [isActive]);

  if (!isActive) return null;
  return <div className={cl(className, styles.overlay)} onClick={close} />;
};
