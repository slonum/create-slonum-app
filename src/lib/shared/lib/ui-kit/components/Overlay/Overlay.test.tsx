import { render } from '@testing-library/react';
import { Overlay } from './Overlay';

describe('Overlay', () => {
  const mockFn = jest.fn();
  it('should render successfully', () => {
    const { container } = render(<Overlay isActive={false} close={mockFn} />);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
  it('active', () => {
    const { container } = render(<Overlay isActive={true} close={mockFn} />);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
