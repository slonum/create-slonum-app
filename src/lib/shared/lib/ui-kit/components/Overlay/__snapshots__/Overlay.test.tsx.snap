// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay active 1`] = `
<div>
  <div
    class="overlay"
  />
</div>
`;

exports[`Overlay should render successfully 1`] = `<div />`;

exports[`Overlay story active 1`] = `
<div>
  <div
    class="overlay story"
  />
</div>
`;

exports[`Overlay story should render successfully 1`] = `<div />`;
